tests/test_raw_image.py

```diff
@@ -15,6 +15,24 @@ def setUp(self):
             for y in range(self.height):
                 self.img.set_pixel(x, y, float(x + y * self.width))
 
+        self.test_gpu = [False]
+        if KB_HAVE_GPU:
+            self.test_gpu.append(True)
+
+    def _do_convolve(self, img, psf, use_gpu):
+        # Make a clean version of the image for the convolution.
+        img2 = raw_image(self.width, self.height)
+        for x in range(self.width):
+            for y in range(self.height):
+                img2.set_pixel(x, y, self.img.get_pixel(x, y))
+
+        if use_gpu:
+            img2.convolve(psf)
+        else:
+            img2.convolve_cpu(psf)
```

Comment:
Let me think about what will be easier to maintain in the long term.

For now, I would like to stay with the `#ifdef` in the C++ code to be consistent with how we are handling the pybind functions. But I agree this could be something we want to change to use templates in a later PR.